src/views/virtualmachinesinstance/details/tabs/disks/utils/virtualMachinesInstancePageDisksTabUtils.ts

```diff
@@ -0,0 +1,38 @@
+import byteSize from 'byte-size';
+
+import { V1Disk } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+
+export type DiskPresentation = {
+  name: string;
+  interface: string;
+  drive: string;
+  source: string;
+  storageClass?: string;
+  size?: string;
+  namespace?: string;
+};
+
+const isDiskOrCdrom = (obj: object) => (obj.hasOwnProperty('disk') ? 'disk' : 'cdrom');
+
+export const convertBytes = (bytes: number) =>
+  byteSize(bytes, {
+    units: 'iec',
+  });
+
+export const diskStructureCreator = (disks: K8sResourceCommon[] | V1Disk[]): DiskPresentation[] => {
```

Comment:
I dont understand why we need a Union Type here.
please create a new type with a disk and a pvc as another field in it

Reply:
regarding https://github.com/kubevirt-ui/kubevirt-plugin/pull/92#discussion_r812867610

Reply:
sure ill create new type